apps/ledger-live-mobile/src/screens/Onboarding/steps/language.tsx

```diff
@@ -97,9 +97,12 @@ function OnboardingStepLanguage({ navigation }: StackScreenProps<{}>) {
 
   // no useCallBack around RNRRestart, or the app might crash.
   const changeLanguageRTL = async () => {
-    await I18nManager.forceRTL(!I18nManager.isRTL);
-    await dispatch(setLanguage(selectedLanguage));
-    RNRestart.Restart();
+    console.log(selectedLanguage);
+    await Promise.all([
+      I18nManager.forceRTL(!I18nManager.isRTL),
+      dispatch(setLanguage(selectedLanguage)),
+    ]);
+    setTimeout(() => RNRestart.Restart(), 0);
```

Comment:
what creates this need? is it the time for the settings changes to be saved in the database? 

we seem to throttle the settings save by 400ms ( https://github.com/LedgerHQ/ledger-live/blob/7921eec738def94d52eb1b57624ed4c68fbad0df/apps/ledger-live-mobile/src/index.tsx#L196 ),
do we need to "flushAll" or an equivalent that would force the save and we await this?

Reply:
It's indeed because of that that we're having that setTimeout. Is there a way to forcefully save the changes in the DB ?
Otherwise, I can add a comment on my code to be clearer on why it's here.

Reply:
yes, there is a `flashAll` function you can call to force and await the effect of saving the DB.

that function is exported from `components/DBSave.ts`